dateparser/utils.py

```diff
@@ -86,3 +86,24 @@ def find_date_separator(format):
     m = re.search(r'(?:(?:%[dbBmaA])(\W))+', format)
     if m:
         return m.group(1)
+
+
+class Registry(type):
```

Comment:
One of alternate implementations would be subclassing and using private attribute in `__new__`. (don't forget here, that `__init__` would still be called).

As for a side note, why do you thing there is **for sure** would be no such temptation? Deprive users of ways to write their own code (however bad it is) is not the best approach in my opinion.